### DISCUSSION TITLE
Migrating Microsoft Journal to Xournal++

### DISCUSSION THREAD

**Discussion by Max0uishere** [xournalpp/xournalpp#3929] [Q&A]
Hello. I want to use Xournal++ but I can't open Microsoft Journal's .*jnt files with it. Is there a way to make them open? If not could you please add this feature?

**Comment by LittleHuba** [comment]
We would at least require a sample file and a screenshot of the file opened in Microsoft Journal. Then we can evaluate whether support of the format is possible.

The sample file should contain any type of element once that is supported by Microsoft Journal.

**Reply by Max0uishere** [reply]
[sample_MS_Journal.zip](https://github.com/xournalpp/xournalpp/files/8372953/sample_MS_Journal.zip)

Hello, I attached the files in this message.

**Reply by rolandlo** [reply]
It looks pretty hard to obtain any information from the jnt file. It's binary data and I find no clue what encryption algorithm is used.

**Reply by Max0uishere** [reply]
I found this script that converts JNT to PDF. Maybe It is possible to integrate this into Xournal++ in order to import .jnt as .pdf?

```
REM Skript sthet unter GPLv3 by jcol11 
REM you need PDFCreator, and Windows to run it

REM for commandline Argument
cd %1

REM create a fresh directory, delete it if it already exists. This should be rewritten. Timestamps should be compared instead of deleting a PDFprint of an unchanged jnt.
echo %cd%
mkdir PDFs_JNT
del /q PDFs_JNT


REM set your default printer, remember this printer
SET %alterdrucker="MYPRINTER"

REM set PDFCreator as your default printer (temporarily)
rundll32 printui.dll,PrintUIEntry /y /n "PDFCreator"
	
REM activate Autosave
REG ADD HKEY_CURRENT_USER\Software\PDFCreator\Program /f /v UseAutosave /d 1

REG ADD HKEY_CURRENT_USER\Software\PDFCreator\Program /f /v Autosavefilename /d print
	
REM Ausgabeverzeichnis setzen	
REG ADD HKEY_CURRENT_USER\Software\PDFCreator\Program /f /v Autosavedirectory /d "%cd%\PDFs_JNT\\"

FOR %%f IN (*.jnt) DO (
	REM PrintFile to pdf
	"%ProgramFiles%\Windows Journal\Journal.exe" /p "%cd%\%%f"
	ping microsoft.com
	ren "%cd%\PDFs_JNT\print.pdf" "%%f.pdf")

REM restore your default printer
rundll32 printui.dll,PrintUIEntry /y /n %alterdrucker%
```

Edit: Source=[https://forum.ubuntuusers.de/topic/ms-journal-dateien-unter-linux-oeffnen-wie/#post-3742637](url)

**Comment by rolandlo** [comment]
Here is a website that converts .jnt to pdf: https://speedtesting.herokuapp.com/jntviewer/
Unfortunately it doesn't seem to read the format on its own.

Here is a link to a [discussion](https://sourceforge.net/p/xournal/discussion/554376/thread/b27001fc/) on the same feature request for Xournal.

**Edit: Updated info**

**Reply by Max0uishere** [reply]
Hello. Thanks I will check the converter.
